src/Raven.Server/Utils/CpuUsage.cs

```diff
@@ -46,24 +48,43 @@ public static (double MachineCpuUsage, double ProcessCpuUsage) Calculate()
             // safe by just holding a lock.
             lock (Locker)
             {
+                (double MachineCpuUsage, double ProcessCpuUsage) cpuInfo;
                 if (PlatformDetails.RunningOnPosix == false)
                 {
-                    var windowsCpuInfo = CalculateWindowsCpuUsage();
-                    _lastCpuInfo = windowsCpuInfo;
-                    return windowsCpuInfo;
+                    cpuInfo = CalculateWindowsCpuUsage();
                 }
-
-                if (PlatformDetails.RunningOnMacOsx)
+                else if (PlatformDetails.RunningOnMacOsx)
+                {
+                    cpuInfo = CalculateMacOsCpuUsage();
+                }
+                else
                 {
-                    var macCpuInfo = CalculateMacOsCpuUsage();
-                    _lastCpuInfo = macCpuInfo;
-                    return macCpuInfo;
+                    cpuInfo = CalculateLinuxCpuUsage();
                 }
 
-                var linuxCpuInfo = CalculateLinuxCpuUsage();
-                _lastCpuInfo = linuxCpuInfo;
-                return linuxCpuInfo;
+                _lastCpuInfo = cpuInfo;
+                return _lastCpuInfo.Value;
+            }
+        }
+
+        private static double CalculateProcessCpuUsage(ProcessInfo currentInfo, ProcessInfo previousInfo, double machineCpuUsage)
+        {
+            var processorTimeDiff = currentInfo.TotalProcessorTimeTicks - previousInfo.TotalProcessorTimeTicks;
+            var timeDiff = currentInfo.TimeTicks - previousInfo.TimeTicks;
+            if (timeDiff <= 0)
+            {
+                //overflow
+                return _lastCpuInfo?.ProcessCpuUsage ?? 0;
+            }
+
+            var processCpuUsage = (processorTimeDiff * 100.0) / timeDiff / currentInfo.ActiveCores;
```

Comment:
Need to check that `ActiveCores` isn't zero either

Reply:
done